addons/common/XEH_postInit.sqf

```diff
@@ -360,6 +376,12 @@ addMissionEventHandler ["PlayerViewChanged", {
     }, {}, [], 3, {TRACE_1("timeout",_this);}] call CBA_fnc_waitUntilAndExecute;
 }];
 
+// block throw when launcher is in hands
+["weapon", {
+    params ["_unit", "_weapon"];
+    [_unit, "blockThrow", QUOTE(ADDON), _weapon == secondaryWeapon _unit] call FUNC(statusEffect_set);
```

Comment:
This will return true when player has no weapon ("" == "")

I also don't like this calling `statusEffect_set` as there are global setVars
We can just directly check this in advThrow/canPrepare

* basically everything else is fairly rare, so it won't really matter, but weapon switching is probably the most common thing to happen

Reply:
I chose `statusEffect` framework exactly because it allows to set effect from several sources at the same time. If we directly check variables then they can be overwritten when 2 sources are changed at the same time.
Though launcher check is really better to check in `advThrow`.
Maybe add `global` parameter to `FUNC(statusEffect_set)` (true by default) and use it here for all calls?